src/mlpack/core/tree/spill_tree/typedef.hpp

```diff
@@ -0,0 +1,119 @@
+/**
+ * @file typedef.hpp
+ * @author Ryan Curtin
+ * @author Marcos Pividori
+ *
+ * Template typedefs for the SpillTree class that satisfy the requirements
+ * of the TreeType policy class.
+ */
+#ifndef MLPACK_CORE_TREE_SPILL_TREE_TYPEDEF_HPP
+#define MLPACK_CORE_TREE_SPILL_TREE_TYPEDEF_HPP
+
+#include "../space_split/mean_space_split.hpp"
+#include "../space_split/midpoint_space_split.hpp"
+
+namespace mlpack {
+namespace tree {
+
+/**
+ * The hybrid spill tree.  It is a variant of metric-trees in which the children
+ * of a node can "spill over" onto each other, and contain shared datapoints.
+ *
+ * When recursively splitting nodes, the SPTree class select the dimension with
+ * maximum width to split on, and picks the midpoint of the range in that
+ * dimension as the value on which to split nodes.
+ *
+ * In each case a "overlapping buffer" is defined, included points at a distance
+ * less than tau from the decision boundary defined by the midpoint.
+ *
+ * For each node, we first split the points considering the overlapping buffer.
+ * If either of its children contains more than rho fraction of the total points
+ * we undo the overlapping splitting. Instead a conventional partition is used.
+ * In this way, we can ensure that each split reduces the number of points of a
+ * node by at least a constant factor.
+ *
+ * For more information, see the following paper.
+ *
+ * @code
+ * @inproceedings{
+ *   author = {Ting Liu, Andrew W. Moore, Alexander Gray and Ke Yang},
+ *   title = {An Investigation of Practical Approximate Nearest Neighbor
+ *     Algorithms},
+ *   booktitle = {Advances in Neural Information Processing Systems 17},
+ *   year = {2005},
+ *   pages = {825--832}
+ * }
+ * @endcode
+ *
+ * This template typedef satisfies the TreeType policy API.
+ *
+ * @see @ref trees, SpillTree, MeanSPTree
+ */
+template<typename MetricType, typename StatisticType, typename MatType>
+using SPTree = SpillTree<MetricType,
+                         StatisticType,
+                         MatType,
+                         AxisOrthogonalHyperplane,
+                         MidpointSpaceSplit>;
```

Comment:
Ah, I see, I guess in that case I disagree with the naming in the paper too, since "SP" is not really an abbreviation.  But since it's what they do in the paper, I guess I am fine with either way, so if you want to change it, feel free, but if not, don't worry about it. :)